blocks/results-list-block/features/results-list/results/index.jsx

```diff
@@ -0,0 +1,347 @@
+      case 'story-feed-sections':
+      case 'story-feed-tag': {
+        return { feedOffset: offset, feedSize: size };
+      }
+      case 'content-api-collections': {
+        return { from: offset, size };
+      }
+      default: { break; }
+    }
+    return {};
+  }, [configuredOffset, configuredSize, contentService]);
+
+  const requestedResultList = useContent({
+    source: isServerSideLazy ? null : contentService,
+    query: {
+      ...contentConfigValues,
+      feature: 'results-list',
+      ...serviceQueryPage(queryOffset),
+    },
+    filter: `{
+      count
+      next
+      content_elements {
+        _id,
+        type
+        display_date
+        credits {
+          by {
+            _id
+            name
+            url
+            type
+            additional_properties {
+              original {
+                byline
+              }
+            }
+          }
+        }
+        headlines {
+          basic
+        }
+        description {
+          basic
+        }
+        promo_items {
+          basic {
+            type
+            url
+            resized_params {
+              274x154
+              158x89
+            }
+          }
+          lead_art {
+            promo_items {
+              basic {
+                type
+                url
+                resized_params {
+                  274x154
+                  158x89
+                }
+              }
+            }
+            type
+          }
+        }
+        websites {
+          ${arcSite} {
+            website_url
+          }
+        }
+      }
+    }`,
+  });
+
+  const [resultList, alterResultList] = useReducer(reduceResultList, requestedResultList);
+
+  useEffect(() => {
+    if (requestedResultList) {
+      alterResultList({
+        type: 'appendUnique',
+        data: requestedResultList,
+      });
+    }
+  }, [requestedResultList]);
+
+  const [elementRefs, setElementRefs] = useState([]);
+  const [focalElement, setFocalElement] = useState(null);
+  useEffect(() => {
+    setElementRefs((existingRefs) => {
+      const refArray = existingRefs.concat(
+        requestedResultList?.content_elements
+          ? requestedResultList.content_elements.map(() => createRef())
+          : [],
+      );
+      if (queryOffset !== configuredOffset) { // ignore the first item for focus purposes
+        const topItem = queryOffset - configuredOffset;
+        if (refArray[topItem]) {
+          setFocalElement(refArray[topItem]);
+        }
+      }
+      return refArray;
+    });
+  }, [configuredOffset, queryOffset, requestedResultList]);
+
+  useEffect(() => {
+    if (focalElement?.current) {
+      const focusLink = focalElement.current.querySelector('a:not([aria-hidden])');
+      if (focusLink) {
+        focusLink.focus();
+      }
+    }
+  }, [focalElement]);
+
+  const [promoElements] = useState(resolveDefaultPromoElements(customFields));
+  const [phrases] = useState(getTranslatedPhrases(locale || 'en'));
```

Comment:
similarly here

Reply:
CC: @matthewroach : This is the main "win" for useState.  These functions (at least one of them) returns an object.  That object is a new object each time the render is called, so it causes the memoized components (and useEffects) to re-render every time even thought he values within don't change.  Since the phrases and the promoElements are not likely to change between renders, making them "state" is a way to prevent re-rendering.

Reply:
so I'm not seeing any effect of phrases on clicking 'read more'. I have been seeing promoElements benefits in usestate for preventing re-renders. it looks like you're using the ref there. Couldn't we use a stable key for this to prevent react from re-rendering the preexisting list items? I can send you the json for the react profiler 

<img width="1138" alt="Screen Shot 2021-07-09 at 10 28 38" src="https://user-images.githubusercontent.com/5950956/125101907-588da880-e0a0-11eb-8746-5a79da8d3126.png">

Reply:
looks like each item is being passed all elements?

Reply:
I was able to not useState for promoElements if I didn't pass each item all elements. I separated out the individual props of showDate, etc

https://github.com/WPMedia/fusion-news-theme-blocks/pull/967/files

<img width="714" alt="Screen Shot 2021-07-09 at 10 38 36" src="https://user-images.githubusercontent.com/5950956/125103599-211ffb80-e0a2-11eb-83ea-c24163e2a324.png">